python/requirements-wheel-test.txt

```diff
@@ -5,16 +5,16 @@ pytest
 pytz
 tzdata; sys_platform == 'win32'
 
-numpy==1.21.3; platform_system == "Linux"   and platform_machine == "aarch64" and python_version < "3.11"
-numpy==1.23.4;                                                                    python_version == "3.11"
-numpy==1.26.0;                                                                    python_version >= "3.12"
+numpy==1.21.6; platform_system == "Linux"   and platform_machine == "aarch64" and python_version < "3.11"
+numpy==1.23.5;                                                                    python_version == "3.11"
+numpy==1.26.4;                                                                    python_version >= "3.12"
```

Comment:
Why do we need these versions?
I think that we want to use the minimum supported NumPy for wheel test to check compatibility.

Reply:
IMO we should simplify this file (especially the differences based on platforms should be able to get removed I think), but let's leave that for a separate issue as that is not really related to this PR.

The changes here just follow the current rationale of this file, which is pinning to the most recent version of numpy for a given platform / python version. In the past, that made sense because we were building against the oldest supported numpy (and so testing with a different, more recent version). Nowadays we are actually building with the most recent version of numpy, and so then it might make sense to test with an older version of numpy. 
But even in that case we still need to have some different versions listed here, as the exact version will probably still depend on the python version.

Reply:
Correct. My rationale was to update to the latest bugfix release for the older supported "major" version.